packages/cosmic-swingset/lib/ag-solo/add-chain.js

```diff
@@ -22,11 +24,46 @@ async function addChain(basedir, chainConfig, force = false) {
     }
   }
 
-  console.log('downloading netconfig from', actualConfig);
-  const r = await fetch(actualConfig);
-  const netconf = await r.json();
+  const url = new URL(actualConfig, `file://${process.cwd()}`);
+  console.log('downloading netconfig from', url.href);
+  let netconf;
+  if (url.protocol === 'file:') {
+    const f = fs.readFileSync(url.pathname, 'utf-8');
+    netconf = JSON.parse(f);
+  } else {
+    const r = await fetch(url.href);
+    netconf = await r.json();
+  }
+  const gciUrl = new URL(netconf.gci, 'hex://');
```

Comment:
Why `hex:` instead of `data:`? Obviously it doesn't matter much, we're the only ones providing the data, but it seems a bit off. I suppose `data:` might also imply `base64`, which could bring up quoting issues..

Reply:
> Why `hex:` instead of `data:`? Obviously it doesn't matter much, we're the only ones providing the data, but it seems a bit off. I suppose `data:` might also imply `base64`, which could bring up quoting issues..

It's not the format of the URL that's supplied, it's just a way of determining when the URL did not specify a scheme.

So I could make it `unspecified://`.

Reply:
Oh, right, the second argument to `new URL` is a referrer. Yeah, use `unknown` or `unspecified`, both sound reasonable to me.